Gifski/Utilities.swift

```diff
@@ -595,6 +595,103 @@ extension AVAsset {
 }
 
 
+extension AVAssetTrack {
+	enum VideoTrimmingError: Error {
+		case assetIsNil
+		case assetReaderFailure
+		case assetIsEmpty
+		case assetIsMissingTrack
+		case compositionCouldNotBeCreated
+	}
+
+	/**
+	Removes blank frames from the begining of the track.
+
+	This can be useful to trim blank frames from files produced by tools like the iOS simulator screen recorder.
+	*/
+	func trimmingBlankFrames() throws -> AVAssetTrack {
+		// Create new composition
+		let composition = AVMutableComposition()
+		guard
+			let wrappedTrack = composition.addMutableTrack(withMediaType: mediaType, preferredTrackID: .zero)
+		else {
+			throw VideoTrimmingError.compositionCouldNotBeCreated
+		}
+		try wrappedTrack.insertTimeRange(timeRange, of: self, at: .zero)
+
+		let reader = try AVAssetReader(asset: composition)
+
+		// Create reader for wrapped track
+		let readerOutput = AVAssetReaderTrackOutput(track: wrappedTrack, outputSettings: nil)
+		reader.add(readerOutput)
+		reader.startReading()
+
+		defer {
+			reader.cancelReading()
+		}
+
+		// Iterate through samples until we reach one with a non-zero size
+		while let sampleBuffer = readerOutput.copyNextSampleBuffer() {
+			guard [.completed, .reading].contains(reader.status) else {
+				throw reader.error ?? VideoTrimmingError.assetReaderFailure
+			}
+
+			// On first non-empty frame
+			guard sampleBuffer.totalSampleSize == 0 else {
+				let currentTimestamp = sampleBuffer.outputPresentationTimeStamp
+
+				wrappedTrack.removeTimeRange(.init(start: .zero, end: currentTimestamp))
+
+				return wrappedTrack
+			}
+		}
+
+		throw VideoTrimmingError.assetIsEmpty
+	}
+}
+
+
+extension AVAssetTrack.VideoTrimmingError: LocalizedError {
+	public var errorDescription: String? {
+		switch self {
+		case .assetIsNil:
+			return "Asset is nil."
+		case .assetReaderFailure:
+			return "Asset could not be read."
+		case .assetIsEmpty:
+			return "Asset is empty."
+		case .assetIsMissingTrack:
+			return "Asset is missing track."
+		case .compositionCouldNotBeCreated:
+			return "Composition could not be created."
+		}
+	}
+}
+
+extension AVAsset {
+	typealias VideoTrimmingError = AVAssetTrack.VideoTrimmingError
+
+	/**
+	Removes blank frames from the begining of the asset.
+
+	This can be useful to trim blank frames from files produced by tools like the iOS simulator screen recorder.
+	*/
+	func trimmingBlankFrames() throws -> AVAsset {
+		guard let videoTrack = self.firstVideoTrack else {
+			throw VideoTrimmingError.assetIsMissingTrack
+		}
+
+		let trimmedTrack = try videoTrack.trimmingBlankFrames()
+
+		guard let trimmedAsset = trimmedTrack.asset else {
+			throw VideoTrimmingError.assetIsNil
```

Comment:
Can this actually happen? I don't think so. I think it would be better to use an `assertionFailure` and just fall back to an empty `AVMutableComposition()`.

Reply:
Probably right, fixed in 9a81320.